cmd/kubeadm/app/componentconfigs/config.go

```diff
@@ -55,6 +65,24 @@ func GetFromKubeletConfigMap(client clientset.Interface, version *version.Versio
 	return obj, nil
 }
 
+// GetKubeletConfigFromFile returns the pointer of the ComponentConfig API object read from the kubelet config file
+func GetKubeletConfigFromFile(kubeletDir string) (runtime.Object, error) {
```

Comment:
i think we can avoid the addition of this function - i.e. unroll it in the single caller.